packages/amplify-prompts/src/validators.ts

```diff
@@ -17,6 +17,11 @@ export const alphanumeric =
   (input: string) =>
     /^[a-zA-Z0-9]+$/.test(input) ? true : message;
 
+export const matchRegex =
+  (message: string = 'Input must satisfy regular expression', validatorRegex : RegExp): Validator =>
```

Comment:
Not sure this default message is very useful. If you feel a default is necessary, perhaps default to: `Input must match regular expression ${validatorRegex}`

Reply:
1. validatorRegex is the second parameter. It cannot be referenced in the default first parameter.
2.  The second parameter to the function is the validatorRegex. The error is saying that the first parameter should satisfy the regex validation. match seems like a better term.

Reply:
You can reference the validatorRegex if you print the message similar to this: https://github.com/aws-amplify/amplify-cli/pull/9555/files/a34ab8f1c50d9494d3bae6c47b26a1b973aa91fd#diff-cc22b4670cb62aadca66250c5c585c16ceb5d570eebd4792f6363540a2df764dR33
To be consistent with the other validators I would make the regex the first input and then the message an optional second parameter

Reply:
Thats a good point, maxLength seems to have the length part first, will make the change to keep it consistent!